flink-runtime/src/main/java/org/apache/flink/runtime/net/SSLUtils.java

```diff
@@ -195,6 +198,22 @@ public static SSLHandlerFactory createRestClientSSLEngineFactory(final Configura
 		return config.getString(SecurityOptions.SSL_ALGORITHMS).split(",");
 	}
 
+	private static SslProvider getSSLProvider(final Configuration config) {
+		checkNotNull(config, "config must not be null");
+		String providerString = config.getString(SecurityOptions.SSL_PROVIDER);
+		if (providerString.equalsIgnoreCase("OPENSSL")) {
+			if (OpenSsl.isAvailable()) {
+				return OPENSSL;
+			} else {
+				return JDK;
+			}
+		} else if (providerString.equalsIgnoreCase("JDK")) {
+			return JDK;
+		} else {
+			throw new IllegalArgumentException("Unknown SSL provider: " + providerString);
```

Comment:
`IllegalArgumentException` runtime exception that doesn't use our exception hierarchy?  Did you mean `IllegalConfigurationException`?

Reply:
sure, why not